app/src/main/java/eu/kanade/tachiyomi/ui/main/MainActivity.kt

```diff
@@ -640,13 +643,23 @@ class MainActivity : BaseActivity() {
                             contains("FFmpeg version:") -> mpvVersions.ffmpeg = this
                             else -> {
                                 recordMPVLog = false
-                                this@MainActivity.onLoggingComplete() // Direct call to MainActivity's method
+                                // Use a safe way to call back to the MainActivity
+                                safeOnLoggingComplete()
                             }
                         }
                     }
                 }
             }
         }
+
+        private fun safeOnLoggingComplete() {
+            // Ensure we're calling MainActivity's method safely
+            (context as? Activity)?.runOnUiThread {
+                if (context is MainActivity) {
+                    (context).onLoggingComplete()
+                }
+            }
```

Comment:
```suggestion
            (context as? MainActivity)?.runOnUiThread {
                context.onLoggingComplete()
            }
```
Can be simplified like this